autoload/leaderf/python/leaderf/commandExpl.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import re
+from .utils import *
+from .explorer import *
+from .manager import *
+
+# :command result line
+# "!|  LeaderfBufTag     0      Leaderf<bang> bufTag"
+#      ^^^^^^^^^^^^^
+RE_USER_DEFINED_COMMAND = re.compile(r"^.{4}(\w+)")
+
+# index.txt line
+# "|:silent|	:sil[ent]	..."
+#    ^^^^^^
+RE_BUILT_IN_COMMAND = re.compile(r"^\|:([^|]+)\|")
+
+#*****************************************************
+# CommandExplorer
+#*****************************************************
+class CommandExplorer(Explorer):
+    def __init__(self):
+        self._content = []
+
+    def getContent(self, *args, **kwargs):
+        if self._content:
+            return self._content
+        else:
+            return self.getFreshContent()
+
+    def getFreshContent(self, *args, **kwargs):
+        result_list = []
+
+        # user-defined Ex commands
+        lfCmd("let tmp = @x")
+        lfCmd("redir @x")
+        lfCmd("silent command")
+
+        result = lfEval("@x")
+
+        lfCmd("let @x = tmp")
+        lfCmd("redir END")
+
+        for line in result.splitlines()[2:]:
+            match = RE_USER_DEFINED_COMMAND.match(line)
+            if match:
+                result_list.append(match.group(1))
+
+        # built-in Ex commands
+        index_file = lfEval("globpath(&rtp, '*/index.txt')")
```

Comment:
Is `index.txt` denoted to `$VIMRUNTIME/doc/index.txt`? If yes, I don't think we need to search `index.txt` in all the directories of &rtp. We can use  `$VIMRUNTIME/doc/index.txt` directly.

Reply:
> Is index.txt denoted to $VIMRUNTIME/doc/index.txt?

Yes.
It is better to specify directly.
I fixed.